handler/src/test/java/io/netty/handler/ssl/SniHandlerTest.java

```diff
@@ -229,4 +249,119 @@ protected void configurePipeline(ChannelHandlerContext ctx, String protocol) thr
             group.shutdownGracefully(0, 0, TimeUnit.MICROSECONDS);
         }
     }
+
+    @Test
```

Comment:
consider adding a timeout to this test ... there are a few places that do `sync` calls without timeouts in this method.